packages/grafana-data/src/types/datasource.ts

```diff
@@ -473,6 +473,13 @@ export interface DataQueryResponse {
   traceIds?: string[];
 }
 
+export interface TestDataSourceResponse {
+  status: string;
+  message: string;
+  error?: Error;
+  details?: { message?: string; verboseMessage?: string };
+}
```

Comment:
@leventebalogh this could lead to new typescript errors in data sources, is that ok in a major version?

Reply:
Good catch. Yeah, I think as it is "only" going to cause a compile-time error, it should break existing plugins, and only needs to be addressed when someone is updating a plugin - which should be fine in a major version I think. What we should probably do is marking it as a breaking change (add it to the CHANGELOG) and mention it in our migration guide.

Reply:
**FYI:** I have added the "breaking change" label and updated the PR description with a breaking change notice message.

Reply:
Thanks for the edits. I think this is a minor breaking change but cleans up the flow a bit in return.